### PR TITLE
bug: disable nftables.service on Azure Linux 2.0

### PR DESCRIPTION
**Pull request by phealy**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
The nftables package was added to Azure Linux to allow for adding rules to fix limitations in how Azure Standard Load Balancer sends health probes for IPv6 services, but when it was added it was missed that (on Azure Linux 2.0) the package enables nftables.service by default, which attempts to load saved nftables rules on boot. However, the default configuration file causes a syntax error, preventing the service from loading properly:

```
root@aks-azurelinux-36603062-vmss000000 [ / ]# systemctl status nftables.service
x nftables.service - Netfilter Tables
     Loaded: loaded (/usr/lib/systemd/system/nftables.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Wed 2024-10-30 16:55:16 UTC; 4 days ago
       Docs: man:nft(8)
   Main PID: 529 (code=exited, status=1/FAILURE)
        CPU: 8ms

Oct 30 16:55:16 localhost.localdomain systemd[1]: Starting Netfilter Tables...
Oct 30 16:55:16 localhost.localdomain nft[529]: /etc/sysconfig/nftables.conf:8:1-1: Error: syntax error, unexpected junk
Oct 30 16:55:16 localhost.localdomain nft[529]: # start by calling: 'nft list ruleset >/etc/sysconfig/nftables.conf'.
Oct 30 16:55:16 localhost.localdomain nft[529]: ^
Oct 30 16:55:16 localhost.localdomain systemd[1]: nftables.service: Main process exited, code=exited, status=1/FAILURE
Oct 30 16:55:16 localhost.localdomain systemd[1]: nftables.service: Failed with result 'exit-code'.
Oct 30 16:55:16 localhost.localdomain systemd[1]: Failed to start Netfilter Tables.
```

This doesn't actually cause a problem on load, but can cause a problem on unload, as the nftables.service file has the following lines:

```
ExecReload=/sbin/nft 'flush ruleset; include "/etc/sysconfig/nftables.conf";'
ExecStop=/sbin/nft flush ruleset
```

This means that if anything triggers a reload of the nftables.service entry, it will flush the nftables ruleset, which can cause issues in a dual-stack cluster. This PR disables the nftables service on Azure Linux 2.0 images to fix this issue.

Azure Linux 3.0 doesn't enable this service, so the change isn't needed there.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
This* have contributed to CRIs in some environments in weird network failure edge cases, but I can't 100% confirm that. I don't think it's critical enough to require urgent merge/hotfixing.

**Release note**:

```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh [hunk 1]
```diff
@@ -19,6 +19,15 @@ installDeps() {
       fi
     done
 
+    # The nftables package turns on a service by default that tries to load config files,
+    # but the stock config files in the package have no uncommented lines and make the service
+    # fail to start. Disabling it as it's not used, and the stop action of "flush tables" can
+    # result in rules getting cleared unexpectedly. Azure Linux 3 fixes this, so we only need
+    # this in 2.0.
+    if [[ $OS_VERSION == "2.0" ]]; then
+      systemctl disable nftables.service || exit 1
```
Comment (djsly) [inline parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh:28]: can we use a specific Error code to avoid confusing with `exit 1`

**Comment by phealy** [comment]
Closing and trying again after completely messing up my local branch...
